<a href="https://colab.research.google.com/github/rahulsait/ExcelR-codes/blob/main/Anova.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Working on "IRIS" Dataset :**

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy
from scipy import stats
%matplotlib inline

In [4]:
# Loading Libraries
from sklearn import datasets
iris = datasets.load_iris()

In [5]:
# Creating a DataFrame :
df = pd.DataFrame(iris.data)

In [6]:
df.head()

0    1    2    3
0  5.1  3.5  1.4  0.2
1  4.9  3.0  1.4  0.2
2  4.7  3.2  1.3  0.2
3  4.6  3.1  1.5  0.2
4  5.0  3.6  1.4  0.2

In [7]:
  iris["feature_names"]

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

In [10]:
# Our Hypothesis is to check wether the values of three columns are alike.
# Ho : Values of All the columns/groups are equal
# Ha : Atleast the value of one group/column is different.

In [9]:
# passing all the 4 columns to ANOVA :
stats.f_oneway(df.iloc[:,0], df.iloc[:,1], df.iloc[:,2], df.iloc[:,3])

F_onewayResult(statistic=482.91531656927964, pvalue=4.660592480454751e-159)

**From the above input, p-value : 4.66 * 10^-159 ~~ 0**

**Since the P-Value (0) is less than < Significance value (alpha = 0.05) we reject the Null Hypothesis**
## **So at least one column has values differing from others.**